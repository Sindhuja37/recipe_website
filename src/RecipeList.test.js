import React from "react";
import { shallow } from "enzyme";
import RecipeList from "./recipeList.js";
import sinon from "sinon";

describe("Test App Entry point", function () {
  it("should have a header tag with Hello world React!", function () {
    const wrapper = shallow(<RecipeList />);
    expect(wrapper.find("h1").text()).toEqual("Tasty recipes");
  });
});

describe("MyComponent", () => {
  it("should render my component", () => {
    const wrapper = shallow(<RecipeList />);
  });
});

describe("MyComponent", () => {
  it("should render initial layout", () => {
    const component = shallow(<RecipeList />);
    expect(component.getElements()).toMatchSnapshot();
  });
});

describe("MyComponent", () => {
  it("responds to input change", (done) => {
    const handleAddRecipe = sinon.spy();
    const actualNode = shallow(<RecipeList onClick={handleAddRecipe} />);
    actualNode.find(".button").simulate("click");
    sinon.assert.called(handleAddRecipe);
  });
});

let wrapper;
const handleNext = jest.fn();
beforeEach(() => {
  wrapper = mount(<ModelStepper handleNext={handleNext} />);
});
it("handleNext increment", () => {
  const btn = wrapper.find(Button).at(1);
  expect(handleNext).toBeCalledTimes(0);
  btn.simulate("click");
  expect(handleNext).toBeCalledTimes(1);
});
