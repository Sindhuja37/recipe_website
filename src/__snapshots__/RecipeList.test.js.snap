// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render initial layout 1`] = `
Array [
  <div>
    <div>
      <h1
        className="app-header"
      >
        Tasty recipes
      </h1>
      <div
        className="autocomplete"
      >
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Recipes"
          type="text"
          value=""
        />
        <ul
          className="autocomplete-list"
        />
        <p>
          Selected Option: 
        </p>
      </div>
    </div>
    <h2
      className="sub-header"
    >
      Recipe List
    </h2>
    <ol
      className="recipe-list"
    />
    <div
      className="add-recipe"
    >
      <h2>
        Add Your Own Recipe
      </h2>
      <div
        className="add-recipe-card"
      >
        <input
          className="recipe-input"
          onChange={[Function]}
          placeholder="Recipe Title"
          type="text"
          value=""
        />
      </div>
      <br />
      <textarea
        className="recipe-input"
        onChange={[Function]}
        placeholder="Ingredients"
        value=""
      />
      <br />
      <textarea
        className="recipe-input"
        onChange={[Function]}
        placeholder="steps"
        value=""
      />
      <br />
      <button
        className="button"
        onClick={[Function]}
      >
        Add Recipe
      </button>
    </div>
  </div>,
]
`;
